### DISCUSSION TITLE
vim equivalent of VS Code's "Expand Selection"?

### DISCUSSION THREAD

**Discussion by valbaca** [VSCodeVim/Vim#9293] [General]
Hi, new to this repo, so sorry if this the wrong place to ask this.

First: LOVE this plugin. Thank you!!! I get all the power of VS Code and Vim. It's awesome.

I'm looking for a VSCodeVim equivalent or near-equivalent of the VS Code "Expand Selection" Ctrl+Cmd+Shift+RightArrow

I find this highlight very useful when moving around code, especially in JS, but it's an awkward key combination.

I figured there would be some kind of `vaf` like but `vaf` is grabbing way more text than "Expand Selection". I want to increase via logical chunks so like:

```
function foo() {
  console.log("blah|")
}
```

I would expand to expand from:

1. blah or "blah"
2. console.log("blah")
3. { ... }
4. function foo() {...}

**Comment by ozkayas** [comment]
for expanding -> enter visual mode "v", then do "af", "af" ... But this is not reversable unfortunately.
Or you can remap VsCode commands with some custom shortcut:
        // {
        //   "before": ["<leader>", "e"],
        //   "commands": ["editor.action.smartSelect.expand"]
        // },
        // {
        //   "before": ["<leader>", "s"],
        //   "commands": ["editor.action.smartSelect.shrink"]
        // }
